apps/myaccount/src/components/shared/system-notification-alert.tsx

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import Alert from "@oxygen-ui/react/Alert";
+import { IdentifiableComponentInterface } from "@wso2is/core/models";
+import React, { FunctionComponent, ReactElement } from "react";
+import "./system-notification-alert.scss";
+import { useResendAccountConfirmationAlert } from "../../hooks/use-resend-account-confirmation-alert";
+import { SystemNotificationAlertState } from "../../models";
+
+/**
+ * SystemNotificationAlert component props.
+ */
+export type SystemNotificationAlertProps = IdentifiableComponentInterface;
+
+/**
+ * System Notification Alert component.
+ *
+ * @param props - Props injected into the component.
+ * @returns ReactElement
+ */
+export const SystemNotificationAlert: FunctionComponent<SystemNotificationAlertProps> = ({
+    ["data-componentid"]: componentId = "system-notification-alert"
+}: SystemNotificationAlertProps): ReactElement | null => {
```

Comment:
With ReactNode, we do not need to specify null in the type

```suggestion
}: SystemNotificationAlertProps): ReactNode => {
```

Reply:
You're correct that `React.ReactNode` encompasses `null`, making it unnecessary to explicitly specify `| null` in the type. However, when defining the return type of a functional component, as per my upnderstanding it's common practice to use `React.ReactElement | null`. This approach ensures that the component returns a valid JSX element or `null`, aligning with React's expectations for component return types.

Using `React.ReactNode` as a return type can introduce ambiguity, as it includes a broader range of types, such as strings, numbers, and booleans, which may not be appropriate in this scenario. Therefore, IMO specifying `React.ReactElement | null` provides clearer intent and better type safety for this component. Please correct me if I am wrong.